#!/usr/bin/env python
PACKAGE = "agv"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("mark_height", double_t, 0, "default: 1.5", 1.5, 0, 3)
gen.add("normal_cruise_speed", double_t, 0, "default: 2.5", 2.5, 1.5, 3)
gen.add("slope_slowdown_speed", double_t, 0, "default: 2", 2, 0, 3)
gen.add("normal_slowdown_speed",double_t, 0, "default: 2", 2, 0, 3)
gen.add("medium_steering_angle_slowdown_speed", double_t, 0, "default: 2.3", 2.3, 0, 3)
gen.add("large_steering_angle_slowdown_speed", double_t, 0, "default: 2", 2, 0, 3)
gen.add("medium_steering_angle", double_t, 0, "default: 0.085", 0.085, 0, 0.336)
gen.add("large_steering_angle", double_t, 0, "default: 0.17", 0.17, 0, 0.336)
gen.add("dynamic_obstacle_brake", double_t, 0, "default: 0.7", 0.7, 0, 1)
gen.add("static_obstacle_brake", double_t, 0, "default: 0.7", 0.7, 0, 1)
gen.add("endpoint_brake", double_t, 0, "default: 0", 0, 0, 1)
gen.add("slope_brake", double_t, 0, "default: 0", 0, 0, 1)
gen.add("dynamic_obstacle_thresh", double_t, 0, "default: 0.3", 0.3, 0, 5)
gen.add("dynamic_obstacle_wait_time", double_t, 0, "default: 3", 3, 0, 5)

exit(gen.generate(PACKAGE, "agv", "behaviour_planner_"))
